src/main/resources/db/migration/V1_68__create_utilisation_report.sql

```diff
@@ -0,0 +1,313 @@
+
+END;
+$function$
+;
+
+
+-- function to return jurors and service within the report window (also indicating public holidays)
+CREATE OR REPLACE FUNCTION juror_mod.util_report_juror_service_list(p_loc_code text, p_start_date date, p_end_date date)
+ RETURNS TABLE(
+ 	juror_number character varying,
+	report_from date,
+	report_to date,
+	service_from date,
+	service_to date,
+	report_date date,
+	within_service boolean,
+	holiday boolean)
+ LANGUAGE plpgsql
+AS $function$
+begin
+
+return query select
+	pool_members.juror_number,
+	p_start_date as report_from,
+	p_end_date as report_to,
+ 	pool_members.service_start_date as service_from,
+ 	pool_members.service_end_date as service_to,
+ 	report_days.report_date,
+	case when report_days.report_date between pool_members.service_start_date and pool_members.service_end_date then true else false end as within_service,
+	report_days.holiday
+	from (select * from juror_mod.util_report_report_days_list(p_loc_code, p_start_date, p_end_date)) as report_days
+	left join (select * from juror_mod.util_report_pool_members_list(p_loc_code, p_start_date, p_end_date)) as pool_members
+	on report_days.report_date between pool_members.service_start_date and pool_members.service_end_date
+	group by  pool_members.juror_number, report_days.report_date, report_days.holiday, pool_members.service_start_date, pool_members.service_end_date
+	order by pool_members.juror_number, report_date;
+END;
+$function$
+;
+
+
+-- function to get the list of jurors in a court within a given date range (used in function above)
+CREATE OR REPLACE FUNCTION juror_mod.util_report_pool_members_list(p_loc_code text, p_start_date date, p_end_date date)
+ RETURNS TABLE(juror_number character varying, attendance_date date, return_date date, service_start_date date, service_end_date date, min_status integer)
+ LANGUAGE plpgsql
+AS $function$
+begin
+
+return query select
+	jp.juror_number,
+    min(a.attendance_date) attendance_date,
+    min(case when jp.is_active = true then p.return_date else to_date('01/12/2099', 'dd/mm/yyyy') end ) return_date,
+    greatest(least(coalesce(min(a.attendance_date), min(case when jp.is_active = true then p.return_date else to_date('01/12/2099', 'dd/mm/yyyy') end )),
+    		min(case when jp.is_active = true then p.return_date else to_date('01/12/2099','dd/mm/yyyy') end )), p_start_date) service_start,
+    least(min(coalesce(case when jp.status = 10 then jp.transfer_date else date(j.completion_date) end, p_end_date)),p_end_date) service_end,
+    min(status) min_status
+    from juror_mod.juror_pool jp
+    join juror_mod.juror j
+    on jp.juror_number = j.juror_number
+    join juror_mod.pool p
+    on jp.pool_number = p.pool_no
+    left join juror_mod.appearance a
+    on jp.juror_number = a.juror_number
+    where p.loc_code = a.loc_code
+    and (((jp.status in (2,3,4,10) or (jp.status = 7 and a.attendance_date is not null)) and jp.is_active ='Y')
```

Comment:
is_active is now a boolean so can be written as `jp.is_active = true`

Reply:
good spot, updated